lib/evm/opFns.js

```diff
@@ -744,18 +740,26 @@ module.exports = {
   // '0x70', range - other
   SELFDESTRUCT: async function (runState) {
     let selfdestructToAddress = runState.stack.pop()
-    if (runState.static) {
+    if (runState.eei.isStatic()) {
       trap(ERROR.STATIC_STATE_CHANGE)
     }
 
+    const balance = await runState.eei.getExternalBalance(runState.eei.getAddress())
+    if (balance.gtn(0)) {
+      const empty = await runState.eei.isAccountEmpty(selfdestructToAddress)
+      if (empty) {
+        runState.eei.useGas(new BN(runState._common.param('gasPrices', 'callNewAccount')))
+      }
+    }
```

Comment:
This is to calculate the gas cost of selfdestruct. It used to be here, but I had moved it to `EEI` to avoid using `stateManager` here. Now I've added the necessary method `isAccountEmpty` to `EEI`, and brought the gas calculation again back to the opcode handler.

Reply:
Ok, thanks.